src/Worker/UpdateServerDirectory.php

```diff
@@ -76,7 +76,7 @@ private static function discoverMastodonDirectory(array $gserver)
 		}
 
 		$accounts = json_decode($result, true);
-		if (empty($accounts)) {
+		if (empty($accounts) || !is_array($accounts)) {
```

Comment:
We can switch the checks to speed this up a little bit.

```suggestion
		if (!is_array($accounts) || empty($accounts)) {
```

Reply:
Doesn't the parser work from left to right and doesn't `is_array` complan when `accounts` is `null`? (I haven't checked)

Reply:
Yes, the parser works from left to right. `!is_array()` will be match earlier on `bool`, `null`, integers or strings. Only if `$accounts` is an array, `empty()` will be called.

See https://3v4l.org/TpKmG

Reply:
another example to make this clear would be:
```suggestion
		if (!is_array($accounts) || count($accounts) === 0) {
```

Reply:
When `is_array` doesn't have got a problem with `null`, then I would simply check for `is_array` since an empty array doesn't cause any problems.

Reply:
If an empty array isn't a problem `!is_array()` is definitely the best option. It only checks, if the (mixed) `$value` is type of array.

```suggestion
		if (!is_array($accounts)) {
```

Reply:
It is changed